.golangci.yml

```diff
@@ -0,0 +1,114 @@
+run:
+  timeout: 5m
+linters:
+# This set of linters are enabled by default: deadcode, errcheck, gosimple, govet, ineffasign, staticcheck, struccheck, typecheck, unused, varcheck
+  enable:
+  # List of all linters: https://golangci-lint.run/usage/linters/
+    - whitespace #https://github.com/ultraware/whitespace
+    - noctx #https://github.com/sonatard/noctx
+    - nilerr #https://github.com/gostaticanalysis/nilerr
+    - nestif #https://github.com/nakabonne/nestif
+    - exportloopref #https://github.com/kyoh86/exportloopref
+    - bodyclose #https://github.com/timakin/bodyclose
+    # - goconst #https://github.com/jgautheron/goconst
+    - stylecheck #https://github.com/dominikh/go-tools/tree/master/stylecheck
+    - revive #golint is deprecated and golangci-lint recommends to use revive instead https://github.com/mgechev/revive
+    #other deprecated lint libraries: maligned, scopelint, interfacer
+    - gocritic #https://github.com/go-critic/go-critic
+    - unparam #https://github.com/mvdan/unparam
+    - misspell #https://github.com/client9/misspell
+    - errorlint #https://github.com/polyfloyd/go-errorlint
+    - gosec #https://github.com/securego/gosec
+    - predeclared #https://github.com/nishanths/predeclared
+    - unconvert #https://github.com/mdempsky/unconvert
+    - wrapcheck #https://github.com/tomarrell/wrapcheck
+issues:
```

Comment:
Pending warnings with logic or complex fixes are listed here. Asana task tickets will be created to fix them on separate PRs to keep the scope small. This PR focuses on the simplest fixes such as `trailing lines`...`unused params`...`spacing in comments` and the likes.